### DISCUSSION TITLE
2024-blog-refactor/

### DISCUSSION THREAD

**Discussion by ghost** [eallion/eallion.com#99] [Giscus]
# 2024-blog-refactor/

副标题：面向死亡编程2023 年结束了！ 2023 年是后疫情时代的第一个年头 2023 年是 AI 元年 2023 年是未来 10 年中最好的一年 这句话明年可以再说一次 但 2023 年，我最大

https://www.eallion.com/2024-blog-refactor/

**Comment by xiewei18** [comment]
赞，喜欢这些思考和见地。细细想来这是个很神奇的事情：具体说喜欢的是哪一部分似乎也说不出来，但总体有很认可。或许这就是互联网脑替。hhh

**Reply by eallion** [reply]
哇喔，谢谢认可，被人认可就是一个神迹。我在上网时也会有这种感觉。

**Comment by binjoo** [comment]
哈哈，之前折腾过这个设计，那个时候不是在折腾hexo嘛，发布分别部署到coding、github，也算是存活了吧。
现在是typecho动态的，也可以思考下如何折腾。。。

**Reply by eallion** [reply]
当初我从 Typecho 转到 Hugo 也是因为有这方面的考虑，Typecho 部署和维护都不难，但也有限制。

**Comment by jacklee981229** [comment]
你好！你的文章很有趣！想問問你把theme放在submodule的思路是怎麽來的？具體怎麽做呢？有沒有什麽reference可以分享呢？

**Reply by eallion** [reply]
现在新推出的 Theme 在它们的文档中都推荐用 Submodule 的方式引入，这样确实好维护和升级。
你可以看看我的 Readme 说明：[https://github.com/eallion/eallion.com#-主题-doit](https://github.com/eallion/eallion.com#-%E4%B8%BB%E9%A2%98-doit)
直接用 Submodule 的方式引入：
```bash
git submodule add https://github.com/eallion/hugo-theme-doit.git themes/DoIt
```
如果对主题有一些自定义的修改，在 Hugo 博客的目录中的 `layouts` 目录下新建同名文件。
比如 
```
https://github.com/eallion/eallion.com/blob/main/layouts/_default/baseof.html
```
就会覆盖掉 
```
https://github.com/eallion/eallion.com/blob/main/thems/DoIt/layouts/_default/baseof.html
```

**Comment by littleyz** [comment]
如果不是新用户，EdgeOne的费用有点小贵。除此，EdgeOne是个好东西。

**Reply by eallion** [reply]
先用一年再看看。
